packages/offix-datastore/src/DataStore.ts

```diff
@@ -1,74 +1,77 @@
+   * If you wish to override adapter you can supply it here
    */
-  clientConfig: GraphQLClientConfig;
+  storeAdapter?: StorageAdapter,
 
   /**
-   * The Schema Version number. Used to trigger a Schema upgrade
+   * Custom replication mechanism that will replicate data.
+   * By default DataStore will be GraphQL (https://graphqlcrud.org) replication mechanism
    */
-  schemaVersion?: number;
+  replicator?: IReplicator
 }
 
+/**
+ * ____ Offix DataStore ___
+ * Client side storage with efficient querying capabilities
+ * and GraphQL replication engine.
+ */
 export class DataStore {
-  public networkStatus: any;
-  private dbName: string;
-  private schemaVersion: number;
   private models: Model<unknown>[];
-  private indexedDB: IndexedDBStorageAdapter;
   private storage: LocalStorage;
-  private clientConfig: any;
-  private metadataName = "metadata";
 
-  constructor(config: DataStoreConfig) {
-    this.dbName = config.dbName;
-    this.schemaVersion = config.schemaVersion || 1; // return 1 if schemaVersion is undefined or 0
-    this.clientConfig = config.clientConfig;
+  private config: DataStoreConfig;
+  private engines: CustomEngines | undefined;
+
+  constructor(config: DataStoreConfig, engines?: CustomEngines) {
+    this.config = config;
     this.models = [];
-    this.indexedDB = new IndexedDBStorageAdapter();
-    this.storage = new LocalStorage(this.indexedDB);
-    this.indexedDB.addStore({ name: this.metadataName });
+    this.engines = engines;
+
+    if (engines && engines.storeAdapter) {
+      this.storage = new LocalStorage(engines.storeAdapter);
+    } else {
+      const indexedDB = new IndexedDBStorageAdapter();
+      this.storage = new LocalStorage(indexedDB);
+    }
+    this.storage.addStore({ name: metadataStoreName });
   }
 
   public createModel<T>(config: ModelConfig<T>) {
-    const model = new Model<T>(config, this.storage, this.metadataName);
+    const model = new Model<T>(config, this.storage, metadataStoreName);
     this.models.push(model);
     return model;
   }
 
   public init() {
-    // TODO rename createGraphQLClient since it is responsible
-    // for the creation of the network status handler too
-    const { gqlClient, networkStatus } = createGraphQLClient(this.clientConfig);
-    this.networkStatus = networkStatus;
-    const queries = buildGraphQLCRUDQueries(this.models);
-    const gqlReplicator = new GraphQLCRUDReplicator(gqlClient, queries);
-    const engine = new MutationReplicationEngine(gqlReplicator, (this.storage as LocalStorage));
-    this.pushReplicator(gqlReplicator);
-    this.indexedDB.createStores(this.dbName, this.schemaVersion);
-    engine.start();
-  }
+    this.storage.createStores(this.config);
 
-  private pushReplicator(replicator: IReplicator) {
-    this.models.forEach((model) => {
-      model.setReplicator(replicator);
-    });
+    if (this.config.replicationConfig) {
```

Comment:
Replication works only when you pass config explicitly. 
Haven't decided it it starts automatically or requires separate method.